### ISSUE TITLE
A few comments on the master branch

### ISSUE THREAD

**Issue by YoshikiOhtani** [cta-observatory/ctapipe_io_magic#27]
- as far as I know, the optics description is a bit different between M1 and M2. When loading MC file with ctapipe.io.EventSource, at least I see the difference of the mirror area:

```
source.subarray.tel[2].optics
OpticsDescription(name=MAGIC, equivalent_focal_length=16.97 m, num_mirrors=1, mirror_area=231.94 m2)

source.subarray.tel[3].optics
OpticsDescription(name=MAGIC, equivalent_focal_length=16.97 m, num_mirrors=1, mirror_area=235.22 m2)
```

- the name of the camera description is now "LSTCam", but should be "MAGICCam".
https://github.com/cta-observatory/ctapipe_io_magic/blob/3e348ef4ce924640df331543beeca2e378ed1852/ctapipe_io_magic/__init__.py#L511
- Is the "use_pedestal" option correctly transferred to the event source? I see the definition lines, but I'm not sure if it is transferred to the valuable self.use_pedestal, does it? Maybe just I don't understand the usage of `ctapipe.core.traits.Bool`?
https://github.com/cta-observatory/ctapipe_io_magic/blob/3e348ef4ce924640df331543beeca2e378ed1852/ctapipe_io_magic/__init__.py#L130
- so if I understood correctly, even if we set `self.use_pedestal = False`, the `MarsCalibratedRun` loads the pedestal events and returned to the event generator, but we only extract the shower events, is it right? Probably better to load only shower events when setting `self.use_pedestal = False` to speed up the loading?

**Comment by aleberti** [comment]
> * as far as I know, the optics description is a bit different between M1 and M2. When loading MC file with ctapipe.io.EventSource, at least I see the difference of the mirror area:
> 
> ```
> source.subarray.tel[2].optics
> OpticsDescription(name=MAGIC, equivalent_focal_length=16.97 m, num_mirrors=1, mirror_area=231.94 m2)
> 
> source.subarray.tel[3].optics
> OpticsDescription(name=MAGIC, equivalent_focal_length=16.97 m, num_mirrors=1, mirror_area=235.22 m2)
> ```

the way the SimTelEventSource creates the CameraDescription, Optics and so on is different from what is done in ctapipe_io_magic. In SimTelEventSource all the information about the telescopes are read from the file itself. In ctapipe_io_magic we rely on the information stored in files. In particular before today we were using:

```python
In [1]: from ctapipe.instrument import OpticsDescription

In [2]: optics = OpticsDescription.from_name("MAGIC")

In [3]: optics
Out[3]: OpticsDescription(name=MAGIC, equivalent_focal_length=16.97 m, num_mirrors=1, mirror_area=239.00 m2)
```

which is unique for both telescopes. Instead possibly in SimTelArray config you have a little different description for M1 and M2. If you want, I can match the one used in SimTelArray, if really needed (in principle not since MC and real data are "separated" and also, you did not have issues with the two different definitions while doing the MAGIC+LST1 analysis)

> 
> * the name of the camera description is now "LSTCam", but should be "MAGICCam".
>   https://github.com/cta-observatory/ctapipe_io_magic/blob/3e348ef4ce924640df331543beeca2e378ed1852/ctapipe_io_magic/__init__.py#L511

true, I will fix it

> * Is the "use_pedestal" option correctly transferred to the event source? I see the definition lines, but I'm not sure if it is transferred to the valuable self.use_pedestal, does it? Maybe just I don't understand the usage of `ctapipe.core.traits.Bool`?
>   https://github.com/cta-observatory/ctapipe_io_magic/blob/3e348ef4ce924640df331543beeca2e378ed1852/ctapipe_io_magic/__init__.py#L130

it is not passed directly to the EventSource base class, which does not have such option, it is something specific of the MAGICEventSource. The option is passed when generating the events: https://github.com/cta-observatory/ctapipe_io_magic/blob/master/ctapipe_io_magic/__init__.py#L766
That way only shower events are returned by the event generator.

> * so if I understood correctly, even if we set `self.use_pedestal = False`, the `MarsCalibratedRun` loads the pedestal events and returned to the event generator, but we only extract the shower events, is it right? Probably better to load only shower events when setting `self.use_pedestal = False` to speed up the loading?

yes, correct. But since number of pedestal events per subrun is quite low wrt shower events, this is not really a bottleneck I would say.

**Comment by YoshikiOhtani** [comment]
OK thanks for your clarification.

> which is unique for both telescopes. Instead possibly in SimTelArray config you have a little different description for M1 and M2. If you want, I can match the one used in SimTelArray, if really needed (in principle not since MC and real data are "separated" and also, you did not have issues with the two different definitions while doing the MAGIC+LST1 analysis)

I think this difference comes from the configuration used in sim_telarray simulation, but I'm also don't know how this difference does affect. Just let's only keep in mind now. 

> true, I will fix it

I already pushed the change directly to master!

> it is not passed directly to the EventSource base class, which does not have such option, it is something specific of the MAGICEventSource. The option is passed when generating the events: https://github.com/cta-observatory/ctapipe_io_magic/blob/master/ctapipe_io_magic/__init__.py#L766
> That way only shower events are returned by the event generator.

Sorry now I understand how it is used. I tested as follows and confirmed the difference:

```
MAGICEventSource(path, use_pedestal=False)  ->  n_events = 13319 (only shower events)
MAGICEventSource(path, use_pedestal=True)  ->  n_events = 1563 (only pedestal events)
```

> yes, correct. But since number of pedestal events per subrun is quite low wrt shower events, this is not really a bottleneck I would say.

OK, since now I see the message about the drive reports 2 times as follows (one for shower events, one for pedestal events), let me modify the logger message a bit. I will push a new commit now.

```
Interpolating events information from 8 drive reports.
Drive reports available from 2020-12-15 21:27:17.971 to 2020-12-15 21:28:21.821.
Interpolating events information from 16 drive reports.
Drive reports available from 2020-12-15 21:27:17.971 to 2020-12-15 21:28:21.821.
```

**Comment by YoshikiOhtani** [comment]
By the way the Cal->DL1 script works fine now. Thanks for the implementation, Alessio!
